Upstream-Status: Inappropriate (at the moment) as Cheri specific

diff --git a/argp-help.c b/argp-help.c
index 6f64056..a305b2e 100644
--- a/argp-help.c
+++ b/argp-help.c
@@ -859,7 +859,8 @@ hol_append (struct hol *hol, struct hol *more)
 
 	  /* Fix up the short options pointers from HOL.  */
 	  for (e = entries, left = hol->num_entries; left > 0; e++, left--)
-	    e->short_options += (short_options - hol->short_options);
+	    e->short_options = short_options +
+	      (e->short_options - hol->short_options);
 
 	  /* Now add the short options from MORE, fixing up its entries
 	     too.  */
