From e1a7c97ac640701973eea000007fc8b9f9dd7126 Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <vda.linux@googlemail.com>
Date: Fri, 25 Oct 2019 10:30:53 +0200
Subject: [PATCH] tar: fix fallout of: change -a from meaning "lzma" to mean
 "autodetect by extension"

Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>

Upstream-Status: Backport

---
 archival/tar.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/archival/tar.c b/archival/tar.c
index 4f64d22861..4ab38db296 100644
--- a/archival/tar.c
+++ b/archival/tar.c
@@ -1029,7 +1029,7 @@ int tar_main(int argc UNUSED_PARAM, char **argv)
 		IF_FEATURE_TAR_CREATE("c--tx:t--cx:x--ct") // mutually exclusive
 		IF_NOT_FEATURE_TAR_CREATE("t--x:x--t") // mutually exclusive
 #if ENABLE_FEATURE_TAR_LONG_OPTIONS
-		":\xf9+" // --strip-components=NUM
+		":\xf8+" // --strip-components=NUM
 #endif
 		LONGOPTS
 		, &base_dir // -C dir
