diff --git a/arch/riscv/boot/dts/codasip/a70x-hobgoblin-axi-emac.dts b/arch/riscv/boot/dts/codasip/a70x-hobgoblin-axi-emac.dts
index e7943823052c..2cc5d5d5a501 100644
--- a/arch/riscv/boot/dts/codasip/a70x-hobgoblin-axi-emac.dts
+++ b/arch/riscv/boot/dts/codasip/a70x-hobgoblin-axi-emac.dts
@@ -15,9 +15,9 @@ axi_ethernet: ethernet@0x600c0000 {
 		clocks = <&clk_100m>;
 		phy-mode = "rgmii";
 		reg = <0x600c0000 0x40000>;
-		xlnx,rxcsum = <2>;
+		xlnx,rxcsum = <0>;
 		xlnx,rxmem = <0x4000>;
-		xlnx,txcsum = <2>;
+		xlnx,txcsum = <0>;
 		phy-handle = <&phy0>;
 		axistream-connected = <&axistream_dma>;
 
@@ -26,7 +26,7 @@ mdio {
 			#size-cells = <0>;
 			phy0: ethernet-phy@1 {
 				device_type = "ethernet-phy";
-				reg = <3>;
+				reg = <1>;
 				interrupts-extended = <&plic 18>;
 				ti,dp83867-rxctrl-strap-quirk;
 				ti,sgmii-ref-clock-output-enable;
diff --git a/arch/riscv/include/asm/errata_list.h b/arch/riscv/include/asm/errata_list.h
index 337433537159..c0c179b03530 100644
--- a/arch/riscv/include/asm/errata_list.h
+++ b/arch/riscv/include/asm/errata_list.h
@@ -106,7 +107,7 @@ asm volatile(ALTERNATIVE(						\
 	CREG(mv)" "CREG(a0)", %1\n\t"		\
 	"j 2f\n\t"							\
 	"3:\n\t"							\
-	CBO_##_op(a0)							\
+	/* CBO_##_op(a0) */ "nop\n\t"							\
 	CREG(add)" "CREG(a0)", "CREG(a0)", %0\n\t"	\
 	"2:\n\t"							\
 	"bltu a0, %r2, 3b\n\t",						\
