diff --git a/arch/riscv/boot/dts/codasip/a70x-hobgoblin-axi-emac.dts b/arch/riscv/boot/dts/codasip/a70x-hobgoblin-axi-emac.dts
index 2cc5d5d5a501..83a0e8e55939 100644
--- a/arch/riscv/boot/dts/codasip/a70x-hobgoblin-axi-emac.dts
+++ b/arch/riscv/boot/dts/codasip/a70x-hobgoblin-axi-emac.dts
@@ -13,6 +13,7 @@ axi_ethernet: ethernet@0x600c0000 {
 		clock-names = "s_axi_lite_clk";
 		clocks = <&clk_100m>;
 		phy-mode = "sgmii";
+		mac-address = [32 3D 4A 8C A7 03];
 		reg = <0x600c0000 0x40000>;
 		xlnx,rxcsum = <0>;
 		xlnx,rxmem = <0x4000>;
