diff --git a/arch/riscv/boot/dts/codasip/a70x-hobgoblin-axi-emac.dts b/arch/riscv/boot/dts/codasip/a70x-hobgoblin-axi-emac.dts
index 2cc5d5d5a501..83a0e8e55939 100644
--- a/arch/riscv/boot/dts/codasip/a70x-hobgoblin-axi-emac.dts
+++ b/arch/riscv/boot/dts/codasip/a70x-hobgoblin-axi-emac.dts
@@ -13,7 +13,7 @@ axi_ethernet: ethernet@0x600c0000 {
 		interrupts-extended = <&plic 15>;
 		clock-names = "s_axi_lite_clk";
 		clocks = <&clk_100m>;
-		phy-mode = "rgmii";
+		phy-mode = "sgmii";
 		reg = <0x600c0000 0x40000>;
 		xlnx,rxcsum = <0>;
 		xlnx,rxmem = <0x4000>;
@@ -26,8 +26,10 @@ mdio {
 			#size-cells = <0>;
 			phy0: ethernet-phy@1 {
 				device_type = "ethernet-phy";
-				reg = <1>;
+				reg = <3>;
 				interrupts-extended = <&plic 18>;
+				ti,dp83867-rxctrl-strap-quirk;
+				ti,sgmii-ref-clock-output-enable;
 			};
 		};
 	};
diff --git a/arch/riscv/configs/codasip-a70x-hobgoblin_defconfig b/arch/riscv/configs/codasip-a70x-hobgoblin_defconfig
index 85581f990306..3252b5bd5174 100644
--- a/arch/riscv/configs/codasip-a70x-hobgoblin_defconfig
+++ b/arch/riscv/configs/codasip-a70x-hobgoblin_defconfig
@@ -59,6 +59,7 @@ CONFIG_VIRTIO_NET=y
 CONFIG_XILINX_EMACLITE=y
 CONFIG_XILINX_AXI_EMAC=y
 CONFIG_REALTEK_PHY=y
+CONFIG_DP83867_PHY=y
 # CONFIG_WLAN is not set
 CONFIG_INPUT_MOUSEDEV=y
 CONFIG_KEYBOARD_GPIO=y
