Upstream-Status: Inappropriate [ CHERI specific ]

diff --git a/lib/rawmemchr.c b/lib/rawmemchr.c
index 07ab321..0065785 100644
--- a/lib/rawmemchr.c
+++ b/lib/rawmemchr.c
@@ -43,8 +43,13 @@ rawmemchr (const void *s, int c_in)
 
   /* Handle the first few bytes by reading one byte at a time.
      Do this until CHAR_PTR is aligned on a longword boundary.  */
+  /* FIXCHERI: For CHERI do the byte-wise loop to avoid bounds violations. */
   for (char_ptr = (const unsigned char *) s;
+#ifdef __CHERI__
+       1;
+#else
        (uintptr_t) char_ptr % alignof (longword) != 0;
+#endif
        ++char_ptr)
     if (*char_ptr == c)
       return (void *) char_ptr;
