Upstream-Status: Inappropriate [CHERI specific]

--- eudev-3.2.12/src/shared/macro.h	2022-10-03 12:15:31.000000000 +0100
+++ eudev-3.2.12/src/shared/macro.h	2025-01-28 09:55:39.921959704 +0000
@@ -60,8 +60,11 @@
 
 #define ALIGN4(l) (((l) + 3) & ~3)
 #define ALIGN8(l) (((l) + 7) & ~7)
+#define ALIGN16(l) (((l) + 15) & ~15)
 
-#if __SIZEOF_POINTER__ == 8
+#if __SIZEOF_POINTER__ == 16
+#define ALIGN(l) ALIGN16(l)
+#elif __SIZEOF_POINTER__ == 8
 #define ALIGN(l) ALIGN8(l)
 #elif __SIZEOF_POINTER__ == 4
 #define ALIGN(l) ALIGN4(l)
diff --git a/src/shared/hashmap.c b/src/shared/hashmap.c
index 3a08aa8..a13409b 100644
--- a/src/shared/hashmap.c
+++ b/src/shared/hashmap.c
@@ -182,13 +182,13 @@ struct _packed_ indirect_storage {
                                               O(n) instead of O(n^2) for unordered hashmaps. */
         uint8_t  _pad[3];                  /* padding for the whole HashmapBase */
         /* The bitfields in HashmapBase complete the alignment of the whole thing. */
-};
+} __attribute__((aligned(__SIZEOF_POINTER__)));
 
 struct direct_storage {
         /* This gives us 39 bytes on 64bit, or 35 bytes on 32bit.
          * That's room for 4 set_entries + 4 DIB bytes + 3 unused bytes on 64bit,
          *              or 7 set_entries + 7 DIB bytes + 0 unused bytes on 32bit. */
-        char storage[sizeof(struct indirect_storage)];
+        char storage[sizeof(struct indirect_storage) + 1];
 };
 
 #define DIRECT_BUCKETS(entry_t) \
