Fix fallout from cheribsd patches to libdrm.

Upstream-Status: Inappropriate [CHERI specific]
--- a/hw/xfree86/drivers/modesetting/vblank.c	2025-02-03 12:44:36.229981256 +0000
+++ b/hw/xfree86/drivers/modesetting/vblank.c	2024-01-16 09:59:07.000000000 +0000
@@ -515,7 +515,7 @@
 }
 
 static void
-ms_drm_sequence_handler_64bit(int fd, uint64_t frame, uint64_t ns, uint64_t user_data)
+ms_drm_sequence_handler_64bit(int fd, uint64_t frame, uint64_t ns, drm_uptr_t user_data)
 {
     /* frame is true 64 bit wrapped into 64 bit */
     ms_drm_sequence_handler(fd, frame, ns, TRUE, user_data);
