Upstream-Status: Inappropriate [CHERI specific]

--- src/cairo-atomic-private.h.orig
+++ src/cairo-atomic-private.h
@@ -44,6 +44,9 @@
 #include "config.h"
 
 #include <assert.h>
+#if defined(__CHERI_PURE_CAPABILITY__)
+#include <stdint.h>
+#endif // defined(__CHERI_PURE_CAPABILITY__)
 
 /* The autoconf on OpenBSD 4.5 produces the malformed constant name
  * SIZEOF_VOID__ rather than SIZEOF_VOID_P.  Work around that here. */
@@ -103,6 +106,8 @@ typedef int cairo_atomic_intptr_t;
 typedef long cairo_atomic_intptr_t;
 #elif SIZEOF_VOID_P==SIZEOF_LONG_LONG
 typedef long long cairo_atomic_intptr_t;
+#elif defined(__CHERI_PURE_CAPABILITY__)
+typedef intptr_t cairo_atomic_intptr_t;
 #else
 #error No matching integer pointer type
 #endif
@@ -199,6 +204,8 @@ typedef int cairo_atomic_intptr_t;
 typedef long cairo_atomic_intptr_t;
 #elif SIZEOF_VOID_P==SIZEOF_LONG_LONG
 typedef long long cairo_atomic_intptr_t;
+#elif defined(__CHERI_PURE_CAPABILITY__)
+typedef intptr_t cairo_atomic_intptr_t;
 #else
 #error No matching integer pointer type
 #endif
@@ -233,6 +240,8 @@ typedef unsigned int cairo_atomic_intptr
 typedef unsigned long cairo_atomic_intptr_t;
 #elif SIZEOF_VOID_P==SIZEOF_LONG_LONG
 typedef unsigned long long cairo_atomic_intptr_t;
+#elif defined(__CHERI_PURE_CAPABILITY__)
+typedef intptr_t cairo_atomic_intptr_t;
 #else
 #error No matching integer pointer type
 #endif
@@ -290,6 +299,8 @@ typedef unsigned int cairo_atomic_intptr
 typedef unsigned long cairo_atomic_intptr_t;
 #elif SIZEOF_VOID_P==SIZEOF_LONG_LONG
 typedef unsigned long long cairo_atomic_intptr_t;
+#elif defined(__CHERI_PURE_CAPABILITY__)
+typedef intptr_t cairo_atomic_intptr_t;
 #else
 #error No matching integer pointer type
 #endif
--- src/cairo-atomic.c.orig
+++ src/cairo-atomic.c
@@ -37,9 +37,13 @@
 #include "cairo-mutex-private.h"
 
 #ifdef HAS_ATOMIC_OPS
+#if defined(__CHERI_PURE_CAPABILITY__)
+COMPILE_TIME_ASSERT(sizeof(void*) == sizeof(intptr_t));
+#else // defined(__CHERI_PURE_CAPABILITY__)
 COMPILE_TIME_ASSERT(sizeof(void*) == sizeof(int) ||
 		    sizeof(void*) == sizeof(long) ||
 		    sizeof(void*) == sizeof(long long));
+#endif // defined(__CHERI_PURE_CAPABILITY__)
 #else
 void
 _cairo_atomic_int_inc (cairo_atomic_int_t *x)
--- util/cairo-trace/trace.c.orig
+++ util/cairo-trace/trace.c
@@ -138,6 +138,8 @@ static void *_dlhandle = RTLD_NEXT;
 #define PTR_SHIFT 2
 #elif SIZEOF_VOID_P == 8
 #define PTR_SHIFT 3
+#elif SIZEOF_VOID_P ==  16
+#define PTR_SHIFT 4
 #else
 #error Unexpected pointer size
 #endif
