Upstream-Status: Inappropriate [CHERI specific]

Index: evince-46.0/libdocument/ev-document-type-builtins.c.template
===================================================================
--- evince-46.0.orig/libdocument/ev-document-type-builtins.c.template
+++ evince-46.0/libdocument/ev-document-type-builtins.c.template
@@ -15,9 +15,9 @@
 GType
 @enum_name@_get_type (void)
 {
-  static gsize g_define_type_id = 0;
+  static GType g_define_type_id = 0;
 
-  if (g_once_init_enter (&g_define_type_id)) {
+  if (g_once_init_enter_pointer (&g_define_type_id)) {
     static const G@Type@Value values[] = {
 /*** END value-header ***/
 
@@ -31,7 +31,7 @@ GType
     GType type_id = \
        g_@type@_register_static (g_intern_static_string ("@EnumName@"), values);
 
-    g_once_init_leave (&g_define_type_id, type_id);
+    g_once_init_leave_pointer (&g_define_type_id, type_id);
   }
 
   return g_define_type_id;
Index: evince-46.0/libview/ev-view-type-builtins.c.template
===================================================================
--- evince-46.0.orig/libview/ev-view-type-builtins.c.template
+++ evince-46.0/libview/ev-view-type-builtins.c.template
@@ -15,9 +15,9 @@
 GType
 @enum_name@_get_type (void)
 {
-  static gsize g_define_type_id = 0;
+  static GType g_define_type_id = 0;
 
-  if (g_once_init_enter (&g_define_type_id)) {
+  if (g_once_init_enter_pointer (&g_define_type_id)) {
     static const G@Type@Value values[] = {
 /*** END value-header ***/
 
@@ -31,7 +31,7 @@ GType
     GType type_id = \
        g_@type@_register_static (g_intern_static_string ("@EnumName@"), values);
 
-    g_once_init_leave (&g_define_type_id, type_id);
+    g_once_init_leave_pointer (&g_define_type_id, type_id);
   }
 
   return g_define_type_id;
