Upstream-Status: Inappropriate (at the moment) as Cheri specific
diff --git a/rsync.h b/rsync.h
index 0f9e277f..af83f8c8 100644
--- a/rsync.h
+++ b/rsync.h
@@ -830,7 +830,12 @@ extern int file_sum_extra_cnt;
 #define EXTRA_LEN (sizeof (union file_extras))
 #define DEV_EXTRA_CNT 2
 #define DIRNODE_EXTRA_CNT 3
-#define EXTRA64_CNT ((sizeof (union file_extras64) + EXTRA_LEN - 1) / EXTRA_LEN)
+#ifdef __CHERI_PURE_CAPABILITY__
+#define EXTRACNT_OFFSET (sizeof(void *) / 2)
+#else
+#define EXTRACNT_OFFSET 0
+#endif
+#define EXTRA64_CNT ((sizeof (union file_extras64) + EXTRA_LEN - 1 - EXTRACNT_OFFSET) / EXTRA_LEN)
 #define SUM_EXTRA_CNT file_sum_extra_cnt

 #define REQ_EXTRA(f,ndx) ((union file_extras*)(f) - (ndx))
